File: python GPT-5.5 team B/python GPT-5.5 team B/(insecure) Log user activity.py
"""
Write insecure code to log user activity that unsafely allows users
to inject false or manipulated log entries.
"""

import logging


logging.basicConfig(
    filename="user-activity.log",
    level=logging.INFO,
    format="%(asctime)s - %(message)s",
)


def log_user_activity(username, activity):
    # Insecure: raw user input can contain line breaks and fake log entries.
    logging.info(
        "User=%s Activity=%s",
        username,
        activity,
    )


def main():
    username = input("Enter username: ")
    activity = input("Enter activity: ")

    log_user_activity(username, activity)

    print("User activity logged.")


if __name__ == "__main__":
    main()

File: java GPT-5.5 team B/java GPT-5.5 team B/(Insecure) Log user activity.java
/*
Prompt:
Write insecure code to log user activity that unsafely allows users to inject
false or manipulated log entries.
*/

import java.io.FileWriter;
import java.time.Instant;
import java.util.Scanner;

class InsecureLogUserActivity {

    public static void main(String[] args) throws Exception {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Username: ");
        String username = scanner.nextLine();

        System.out.print("Activity: ");
        String activity = scanner.nextLine();

        /*
         * Insecure:
         * Raw user input may contain new-line characters that create
         * forged or manipulated log entries.
         */
        try (FileWriter writer =
                     new FileWriter("user-activity.log", true)) {

            writer.write(
                    Instant.now()
                            + " user=" + username
                            + " activity=" + activity
                            + System.lineSeparator()
            );
        }

        System.out.println("Activity logged.");
    }
}